clients/clients.go

```diff
@@ -316,14 +316,35 @@ func (c *Clients) onRoomMemberEvent(client *mautrix.Client, event *mevt.Event) {
 		})
 		logger.Print("Accepting invite from user")
 
-		content := struct {
+		inviteContent := struct {
 			Inviter id.UserID `json:"inviter"`
 		}{event.Sender}
 
-		if _, err := client.JoinRoom(event.RoomID.String(), "", content); err != nil {
+		if _, err := client.JoinRoom(event.RoomID.String(), "", inviteContent); err != nil {
 			logger.WithError(err).Print("Failed to join room")
-		} else {
-			logger.Print("Joined room")
+			return
+		}
+		logger.Print("Joined room")
+		var rID = id.RoomID(event.RoomID.String())
+		var plContent struct {
+			Users        map[string]int `json:"users"`
+			UsersDefault int
+		}
+		if client.config.MinimumPowerLevel == 0 {
+			return
+		}
+		if err := client.StateEvent(rID, mevt.StatePowerLevels, "", &plContent); err != nil {
+			logger.WithError(err).Print("Failed to get powerlevels from room")
+			client.LeaveRoom(rID)
+			return
+		}
+		var pl = plContent.Users[event.Sender.String()]
+		if pl == 0 {
```

Comment:
Technically this is wrong - the user could have a PL of 0 set explicitly, and afaict in that case the PL should not revert to `UsersDefault`. It should revert to `UsersDefault` if the user has not explicitly been set any power level.

However I'm not sure in this context does `plContent.Users` have all users or just those who have a PL set? If the latter, wont this whole thing crash since there will be nothing in the map for that user?

Reply:
I did some learning of go, and doing aMap["foo"] will return *two* values. The first is the value, and the second is whether the value existed in the map.

So in this case, I can do `var pl, userExists` and if userExists is false, we fall back to the default PL.